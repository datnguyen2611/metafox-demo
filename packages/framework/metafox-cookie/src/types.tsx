import { CookieAttributes } from 'js-cookie';

export interface CookieBackendConfig extends CookieAttributes {
  prefix: string;
  attributes?: CookieAttributes;
}
