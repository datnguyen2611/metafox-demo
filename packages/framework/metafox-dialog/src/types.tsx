import { ButtonProps, DialogProps } from '@mui/material';

export interface ModalItemParams<P = Record<string, any>> {
  component: string | React.ElementType;
  dialogId?: string;
  permanent?: boolean;
  open?: boolean;
  props?: P;
}

export interface TDialogProvider {
  content: { component: string; props: Record<string, any> };
  type: string;
  resolve(value: any): void;
  isLast: boolean;
  onClose?: () => void;
  forceClose?: () => void;
  open?: boolean;
  onEnter?: () => void;
  onEntering?: () => void;
  onEntered?: () => void;
  onExit?: () => void;
  onExiting?: () => void;
  onExited?: () => void;
  backdropClose?: boolean;
}

interface AlertButtonProps {
  label: string;
  color?: ButtonProps['color'];
  variant?: ButtonProps['variant'];
  size?: ButtonProps['size'];
}

export interface DialogBackendConfig {}

export interface AlertParams {
  title?: React.ReactNode;
  message?: React.ReactNode;
  positiveButton?: AlertButtonProps;
  maxWidth?: DialogProps['maxWidth'];
  disableClose?: boolean;
  showDialogActions?: boolean;
}

export interface TModalDialogProps {
  open: boolean;
  onClose(): void;
  onExited(): void;
  onEnter?(): void;
  onEntered?(): void;
  onEntering?(): void;
  onEscapeKeyDown?(): void;
  onExit?(): void;
  onExiting?(): void;
}
export interface AlertDialogProps extends AlertParams, TModalDialogProps {}

export interface ConfirmParams {
  title?: React.ReactNode;
  message: string | React.ReactNode;
  positiveButton?: AlertButtonProps;
  negativeButton?: AlertButtonProps;
  phraseParams?: Record<string, any>;
}
export type UserConfirmCallback =
  | (() => ConfirmParams | boolean | undefined)
  | boolean
  | undefined;

export interface DialogItemContext {
  type: string;
  dialogProps?: DialogProps;
  closeDialog: () => void;
  forceClose: () => void;
  setDialogValue: (value: unknown) => void;
  setDialogResolveValue: (value: unknown) => void;
  setUserConfirm: (
    callback?: UserConfirmCallback,
    shouldCheckNavigation?: boolean
  ) => void;
  disableBackdropClick: (disabled: boolean) => void;
}
