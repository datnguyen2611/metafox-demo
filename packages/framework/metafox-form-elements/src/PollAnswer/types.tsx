export interface AnswerPollProps {
  lastElement?: boolean;
  item: AnswerItemProps;
  handleDelete?: (order?: number) => void;
  handleAdd?: () => void;
  handleChange?: (e, id: number) => void;
  submitCount?: number;
  index?: number;
  error?: string;
  disabled?: boolean; // default false
  maxLength?: number;
  disabledAddMore?: boolean;
  disableRemove?: boolean;
  meta?: Record<string, any>;
  setTouched: (value: boolean, shouldValidate?: boolean) => void;
}

export interface AnswerItemProps {
  id?: number;
  order?: number;
  answer: string;
}
