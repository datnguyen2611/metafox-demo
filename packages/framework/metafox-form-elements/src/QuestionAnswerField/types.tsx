export interface AnswerQuizProps {
  answer: string;
  answer_id?: number;
  is_correct?: 0 | 1;
  ordering?: number;
}

export interface QuestionItemProps {
  question: string;
  question_id?: number;
  answers: AnswerQuizProps[];
  ordering?: number;
}

export interface QuizItemProps {
  index: number;
  name: string;
  disabled: boolean;
  item: QuestionItemProps;
  lastElement: boolean;
  maxLength?: any;
  formik: any;
  error: string | Record<string, any>;
  submitCount?: number;
  handleChangeAnswer: (e, questionIndex: number, answerIndex: number) => void;
  handleChangeQuestion: (e, index: number) => void;
  addMoreQuestion: () => void;
  handleDeleteQuestion: () => void;
  addMoreAnswer: () => void;
  handleDeleteAnswer: (value) => void;
  handleChangeCorrectAnswer: (
    e,
    questionIndex: number,
    answerIndex: number
  ) => void;
}
