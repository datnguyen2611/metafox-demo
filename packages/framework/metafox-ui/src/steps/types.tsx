import { RenderItemBase } from '@metafox/framework/types';

export interface StepShape extends RenderItemBase {
  title: string;
  id: string;
  expanded?: boolean;
  completed?: boolean;
  disabled?: boolean;
}

export interface State {
  activeStep: number;
  activeId?: string;
  data: { steps: StepShape[] };
}

export interface StepsProps {
  steps: StepShape[];
}

export interface ProcessStepShape {
  title: string;
  enableReport?: boolean;
  disableRetry?: boolean;
  dryRun?: boolean;
  disableUserAbort?: boolean;
  timeout?: number;
  dataSource: { apiUrl: string; apiMethod?: string };
  data?: object; // passed to api request for POST, PUT
  params?: object; // passed to api request for GET, DELETE
  message?: string;
}
export interface ProcessState {
  steps: StepShape[];
  current: number;
  done?: boolean;
}

export interface AppItemShape {
  identity: string;
  name: string;
  version: string;
}
