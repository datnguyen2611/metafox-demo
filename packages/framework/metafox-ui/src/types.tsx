import type {
  GridItemProps,
  HandleAction,
  RemoteDataSource
} from '@metafox/framework/types';
import { ThemeId } from '@metafox/layout';
import {
  ButtonProps,
  GridProps,
  IconButtonProps,
  PopperProps,
  SxProps,
  TypographyProps,
  TypographyVariant
} from '@mui/material';
import { DataGridProps } from '@mui/x-data-grid';
import { LinkProps } from 'react-router-dom';

export type GridColumns = DataGridProps['columns'];

export type ItemItemPropsShape = Record<string, any>;

export type ItemStatisticShape = {
  total_like: number;
  total_comment: number;
  total_score: number;
  total_rating: number;
  [key: string]: number;
  total_item: number;
  total_photo: number;
};

export interface FeedEmbedCardProps<A = any> {
  variant?: 'grid' | 'list' | 'default';
  bottomSpacing?: 'normal' | 'dense';
  children?: React.ReactNode;
  link?: string;
  item: A;
  feed: any;
  isShared?: boolean;
  sxOuter?: SxProps;
  sx?: SxProps;
}

export interface FeedEmbedCardMediaProps {
  image?: string;
  widthImage?: string;
  heightImage?: string;
  mediaRatio?: ImageRatio;
  variant?: 'grid' | 'list' | 'default';
  link?: string;
  playerOverlay?: boolean;
  playerOverlayProps?: TImagePlayerOverlay;
  host?: string;
}

export type TLineIconVariant = 'listItemIcon' | 'itemActionIcon';

export interface LineIconProps {
  icon: string;
  component?: React.ElementType;
  className?: string;
  variant?: TLineIconVariant;
  color?:
    | 'primary'
    | 'secondary'
    | 'textPrimary'
    | 'textSecondary'
    | 'inherit'
    | string;
  onClick?: (e?: any) => void;
  style?: any;
  role?: string;
  sx?: SxProps;
}

export type ItemExtraShape = {
  can_view: boolean;
  can_like: boolean;
  can_share: boolean;
  can_delete: boolean;
  can_delete_own: boolean;
  can_download: boolean;
  can_report: boolean;
  can_add: boolean;
  can_edit: boolean;
  can_comment: boolean;
  can_feature: boolean;
  can_approve: boolean;
  can_sponsor: boolean;
  can_sponsor_in_feed: boolean;
  can_purchase_sponsor: boolean;
  can_unlike: boolean;
  can_start_discussion: boolean;
  can_change_privacy_from_feed?: boolean;
  can_edit_cover_photo?: boolean;
  can_message?: boolean;
  can_upload_media?: boolean;
  can_add_cover?: boolean;
  can_remove_link_preview?: boolean;
  can_close?: boolean;
  can_show_payment_button?: boolean;
  can_show_no_payment_gateway_message?: boolean;
  can_view_comment?: boolean;
  can_show_sponsor_label?: boolean;
};

export type BackgroundSize = '48' | '300' | '1024' | 'origin';

export type AvatarSize = '50' | '120' | '200' | 'origin';

export type CoverSize =
  | '75'
  | '100'
  | '150'
  | '240'
  | '500'
  | '1024'
  | 'origin';

export interface ItemUserShape {
  resource_name: string;
  module_name: string;
  full_name: string;
  avatar: Record<AvatarSize, string>;
  link: string;
  is_featured: boolean;
  friend_id: number;
  user_name: string;
  id: number;
  cover: Record<CoverSize, string>;
  statistic: ItemStatisticShape;
  extra?: ItemExtraShape;
  city_location?: string;
  joined?: string;
}

export type ItemShape = {
  id: number | string;
  module_id: string;
  resource_name: string;
  is_featured: boolean;
  is_sponsor: boolean;
  is_sponsored_feed: boolean;
  is_liked: boolean;
  is_pending: boolean;
  is_online: boolean;
  item_id: number;
  creation_date: string;
  image?: Record<string, string>;
  images?: Record<string, string>[];
  extra?: ItemExtraShape;
  user: string;
  owner: string;
  price?: string;
  text_parsed?: string;
  statistic?: ItemStatisticShape;
  link?: string;
  url?: string;
  currency_id?: string;
  short_description?: string;
  description?: string;
  country?: string;
  city?: string;
  province?: string;
  privacy?: number;
  privacy_feed?: {
    icon: string;
    tooltip: string;
    label?: string;
  };
  privacy_detail?: {
    icon: string;
    tooltip: string;
    label?: string;
  };
  summary?: string;
  weblink?: string;
  attachments: string[];
  _identity: string;
};

export type SimpleUserShape = {
  id: number;
  user_name: string;
  full_name: string;
  avatar: string;
  friendship?: number;
  module_name?: string;
  link?: string;
};

export type ItemViewProps<T, U = unknown, S = unknown> = {
  identity: string;
  item: T;
  user?: SimpleUserShape & ItemUserShape;
  itemProps: ItemItemPropsShape;
  actions: U;
  state: S;
  setState: (cb: (prev: S) => S) => void;
  handleAction: (type: string, payload?: unknown) => void;
  wrapAs: React.FC<GridItemProps>;
  wrapProps: GridItemProps;
  pagingId?: string;
};

export type DetailViewProps<T, U = unknown, S = unknown> = {
  identity: string;
  item: T;
  user?: SimpleUserShape;
  actions: U;
  state: S;
  setState: (cb: (prev: S) => S) => void;
  handleAction: (type: string, payload?: unknown) => void;
};

export interface EmbedObjectShape {
  id: number | string;
  module_id: string;
  resource_name?: string;
  module_name?: string;
  is_featured: boolean;
  is_sponsor: boolean;
  is_liked: boolean;
  is_pending: boolean;
  item_id: number;
  creation_date: string;
  image?: string | Record<string, string>;
  images?: Record<string, string>;
  statistic: ItemStatisticShape;
  extra: ItemExtraShape;
  user: ItemUserShape;
  price?: string;
  link?: string;
  currency_id?: string;
  short_description?: string;
  country?: string;
  city?: string;
  province?: string;
  summary?: string;
  title?: string;
  description?: string;
}

export type EmbedItemInFeedItemProps<A = any> = {
  item: A;
  feed: any;
  isShared?: boolean;
  itemView?: string;
  photo: Record<string, any>;
};

export interface FeedEmbedObjectViewProps<
  A extends EmbedObjectShape = EmbedObjectShape,
  B = ItemShape
> {
  embed: A;
  feed: B;
  itemView?: string;
}

export type ItemEmbedObjectShape = ItemShape;

export interface UIBlockProps {
  variant?: string;
  noHeader?: boolean;
  noFooter?: boolean;
  [key: string]: any;
}

export interface UIGridContainerProps extends GridProps {}

export interface UIGridItemProps extends GridProps {}

export interface ItemTabProps {
  id?: string;
  tab: string;
  label: string;
  showWhen?: Record<string, any>;
  statisticKey?: string;
  [key: string]: any;
}

export interface UIBlockViewProps<A = RemoteDataSource, B = UIBlockProps> {
  title?: string;
  themeId?: ThemeId;
  tabProps?: {
    tabs: ItemTabProps[];
    activeTab: string;
    disableGutter: boolean;
    tabsNoSearchBox?: string[];
    placeholderSearch?: string;
    to?: string;
  };
  elements: any;
  dataSource?: A;
  blockView?: string;
  blockHeader?: string;
  blockFooter?: string;
  blockContent?: string;
  blockProps?: B;
}

export type ItemActionMenuProps = {
  id?: string;
  testid?: string;
  label?: string;
  identity?: string;
  menuName?: string;
  dependName?: string;
  appName?: string;
  icon?: string;
  keepMounted?: boolean;
  disableRipple?: boolean;
  className?: string;
  iconClassName?: string;
  children?: React.ReactNode;
  handleAction?: HandleAction;
  size?: ButtonProps['size'];
  color?: IconButtonProps['color'];
  variant?: IconButtonProps['variant'];
  // state?: any; // remove state pass from items.
  control?: JSX.Element;
  scrollRef?: React.RefObject<HTMLDivElement>;
  scrollClose?: boolean;
  placement?: PopperProps['placement'];
  items?: MenuItemShape[];
  disablePortal?: boolean;
  prefixName?: string;
  fallbackName?: string;
  tooltipTitle?: string;
  preventDefault?: boolean;
  sx?: SxProps;
  autoHide?: boolean;
  preventClose?: boolean;
  zIndex?: number;
  popperOptions?: any;
  popperStyles?: Record<string, any>;
  menuStyles?: Record<string, any>;
  variantPopper?: 'hidden-outview' | string;
  triggerOpen?: any;
};

export type MenuItemValue = string;

export interface MenuItemShape {
  className?: string;
  as?: 'link' | 'button' | 'divider' | 'popover' | 'heading' | string;
  note?: string;
  role?: string;
  target?: '_blank' | '_self' | '_parent' | '_top' | string;
  rel?: 'string';
  image?: string;
  dense?: boolean;
  alt?: string;
  active?: boolean;
  onClick?(): void;
  icon?: string;
  iconSecondary?: string;
  label?: string;
  labelSub?: string;
  value?: MenuItemValue;
  params?: Record<string, any>;
  showWhen?: any;
  enabledWhen?: any;
  variant?: 'text' | 'contained' | 'outlined';
  color?: string;
  disabled?: boolean;
  iconVariant?: any;
  testid?: string;
  to?: string;
  asModal?: boolean;
  asChildPage?: boolean;
  keepScroll?: boolean;
  item_name?: string;
  name?: string;
  buttonProps?: Record<string, any>;
  dividerProps?: {
    variant?: 'inset' | 'fullWidth' | 'middle';
    light?: boolean;
    className?: string;
  };
  placement?: PopperProps['placement'];
  items?: MenuItemShape[];
  order_by?: string;
  content?: {
    component: string | React.ElementType;
    props?: Record<string, any>;
  };
  tab?: string;
  preventClose?: boolean;
  dataSource?: { apiUrl?: string; apiParams?: string };
  behavior?: 'more' | 'last' | 'close';
  disablePortal?: boolean;
  menuItemType?: string;
}

export interface MenuItemViewProps {
  item: MenuItemShape;
  classes: Record<string, any>;
  handleAction?: HandleAction;
  pathname?: string;
  active?: boolean;
  handleCheckActiveMenu?: (url: string, path: string) => boolean;
  minimize?: boolean;
  isDesktop?: boolean;
  activeHover?: boolean;
  handlePopoverClose?: () => void;
  handlePopoverOpen?: () => void;
}

export interface MenuItemProps extends MenuItemShape {
  children?: JSX.Element;
  handleAction?: any;
  classes?: Record<
    'menuItem' | 'activeMenuItem' | 'menuLink' | 'menuItemIcon' | string,
    string
  >;
}

export type ImageRatio =
  | 'auto'
  | 'fixed'
  | '169'
  | '32'
  | '23'
  | '43'
  | '34'
  | '13'
  | '11';

export type ItemReactionShape = {
  resource_name: string;
  module_name: string;
  title: string;
  icon: string;
  color: string;
  ordering: number;
  is_default: boolean;
  id: number;
};

export type ItemCommentShape = {
  user: ItemUserShape;
  text: string;
  parent_id: string;
  extra: ItemExtraShape;
  creation_date: string;
  children?: string[];
  statistic: ItemStatisticShape;
  most_reactions?: string[];
  user_reacted?: string;
  extra_data?: Record<string, any>;
};

export type ActionCommentShape = {
  statistic: ItemStatisticShape;
  most_reactions?: Array<ItemReactionShape>;
};

export type TImagePlayerOverlay = {
  icon?: string;
  size?: 'xs' | 'sm' | 'md' | 'lg';
};
export enum ImageMatureState {
  Normal = 0,
  Warning = 1,
  Strict = 2
}

export type ImageShape = {
  identity?: string;
  classes?: Record<string, any>;
  alt?: string;
  src: string;
  aspectRatio?: ImageRatio;
  shape?: 'square' | 'circle' | 'radius';
  link?: LinkProps['to'];
  host?: string;
  asModal?: boolean;
  backgroundImage?: boolean;
  playerOverlay?: boolean;
  imageFit?: 'cover' | 'contain';
  playerOverlayProps?: TImagePlayerOverlay;
  className?: string;
  imgClass?: string;
  draggable?: boolean;
  'data-testid'?: string;
  style?: Record<string, string>;
  mature?: ImageMatureState;
  linkParams?: Record<string, string>;
  matureProps?: Record<string, any>;
  onLoad?: () => void;
};

export interface TruncateTextProps extends TypographyProps {
  component?: React.ElementType;
  lines?: 0 | 1 | 2 | 3 | 4 | 5;
  fontWeight?: number;
  fixHeight?: boolean;
  showFull?: boolean;
  isIE?: boolean;
  notSupportLineClamp?: boolean;
  isSafari?: boolean;
}

export type ControlMenuItemProps = {
  item: MenuItemShape;
  handleAction?: HandleAction;
  variant?: TypographyVariant;
  as: React.ElementType;
  classes?: Record<string, string>;
  onClick?: () => void;
  className?: string;
  closeMenu?: () => void;
  active?: boolean;
};

export type TotalCommentProps = {
  total: number;
  handleAction?: HandleAction;
  message: string;
  identity: string;
  dataSource?: RemoteDataSource;
};
