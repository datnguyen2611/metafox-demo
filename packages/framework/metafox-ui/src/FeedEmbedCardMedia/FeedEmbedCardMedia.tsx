import { FeedEmbedCardMediaProps, Image } from '@metafox/ui';
import { getImageSrc } from '@metafox/utils';
import { styled, Box } from '@mui/material';
import React from 'react';

const Root = styled(Box, {
  name: 'MuiFeedEmbedCardMedia',
  slot: 'bgCover',
  shouldForwardProp: prop => prop !== 'widthImage' && prop !== 'heightImage'
})<{ widthImage?: string; heightImage?: string }>(
  ({ theme, widthImage, heightImage }) => ({
    width: widthImage,
    height: heightImage,
    '& img': {
      border: '1px solid transparent',
      borderRightColor: theme.palette.border?.secondary,
      borderTop: 'none',
      borderBottom: 'none',
      borderLeft: 'none'
    }
  })
);

export default function FeedEmbedCardMedia({
  image,
  widthImage = '200px',
  heightImage = 'auto',
  mediaRatio = '11',
  link,
  playerOverlay = false,
  playerOverlayProps = {},
  host,
  sx
}: FeedEmbedCardMediaProps) {
  return (
    <Root
      widthImage={widthImage}
      heightImage={heightImage}
      sx={sx}
      className={'media'}
    >
      <Image
        link={link}
        src={getImageSrc(image)}
        aspectRatio={mediaRatio}
        playerOverlay={playerOverlay}
        playerOverlayProps={playerOverlayProps}
      />
    </Root>
  );
}
