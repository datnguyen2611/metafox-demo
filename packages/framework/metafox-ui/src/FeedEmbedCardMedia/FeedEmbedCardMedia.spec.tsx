import { GlobalProvider, Manager } from '@metafox/framework';
import { IntlProvider } from '@metafox/intl';
import { ThemeProvider } from '@metafox/layout';
import { render } from '@testing-library/react';
import * as React from 'react';
import { MemoryRouter } from 'react-router';
import { FeedEmbedCardMediaProps } from '../types';
import FeedEmbedCardMedia from './FeedEmbedCardMedia';

const MockInfoLoader = () => <div data-testid="infoLoader" />;

describe('FeedEmbedCardMedia', () => {
  const manager = Manager.factory({});
  const usePreference = jest.fn().mockImplementation(() => ({}));
  const getConfig = jest.fn().mockImplementation(() => {});
  const mockOn = jest.fn();
  const mockOff = jest.fn();

  manager.use({
    usePreference,
    eventCenter: { on: mockOn, off: mockOff },
    constants: { isLayoutPreviewWindow: false },
    getConfig
  });

  it('render', async () => {
    const { container } = render(
      <GlobalProvider value={manager}>
        <IntlProvider>
          <ThemeProvider infoLoader={MockInfoLoader}>
            <FeedEmbedCardMedia />
          </ThemeProvider>
        </IntlProvider>
      </GlobalProvider>
    );

    expect(container).toMatchSnapshot();
  });
  it('render have link, host props', async () => {
    const props: FeedEmbedCardMediaProps = {
      host: 'host',
      link: '/home'
    };
    const { container } = render(
      <GlobalProvider value={manager}>
        <IntlProvider>
          <ThemeProvider infoLoader={MockInfoLoader}>
            <MemoryRouter>
              <FeedEmbedCardMedia {...props} />
            </MemoryRouter>
          </ThemeProvider>
        </IntlProvider>
      </GlobalProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
