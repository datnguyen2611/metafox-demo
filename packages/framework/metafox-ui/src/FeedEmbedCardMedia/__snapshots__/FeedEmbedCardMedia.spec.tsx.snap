// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedEmbedCardMedia render 1`] = `
<div>
  <div
    data-testid="infoLoader"
  />
  <div
    class="MuiFeedEmbedCardMedia-root media ltr-kvqzwk-MuiFeedEmbedCardMedia-bgCover"
  >
    <div
      class="MuiImage-root MuiImage-square MuiImage-ratio11 MuiImage-cover"
    >
      <div
        class="MuiImage-inner"
      >
        <img
          draggable="false"
          src="[object Object]"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`FeedEmbedCardMedia render have link, host props 1`] = `
<div>
  <div
    data-testid="infoLoader"
  />
  <div
    class="MuiFeedEmbedCardMedia-root media ltr-kvqzwk-MuiFeedEmbedCardMedia-bgCover"
  >
    <a
      class="MuiImage-root MuiImage-square MuiImage-ratio11 MuiImage-cover"
      href="/home"
      target="_blank"
    >
      <div
        class="MuiImage-inner"
      >
        <img
          draggable="false"
          src="[object Object]"
        />
      </div>
    </a>
  </div>
</div>
`;
